src/dotnet-interactive/CommandLine/CommandLineParser.cs

```diff
@@ -291,30 +265,51 @@ Command StdIO()
                     "Starts dotnet-interactive with kernel functionality exposed over standard I/O")
                 {
                     defaultKernelOption,
-                    logPathOption,
-                    httpPortOption,
-                    httpPortRangeOption
+                    logPathOption
                 };
 
                 command.Handler = CommandHandler.Create<StartupOptions, StdIOOptions, IConsole, InvocationContext>(
-                    (startupOptions, options, console, context) =>
-                    {
-                        if (startupOptions.EnableHttpApi)
-                        {
-                            RegisterKernelInServiceCollection(services, startupOptions, options.DefaultKernel,
-                                kernel => StdIOCommand.CreateServer(kernel, console));
-                            startServer?.Invoke(startupOptions, context);
-                            return Task.FromResult(0);
-                        }
-
-                        return startStdIO(
-                            startupOptions,
-                            CreateKernel(options.DefaultKernel, new BrowserFrontendEnvironment(), startupOptions, null),
-                            console);
-                    });
+                    (startupOptions, options, console, context) => startStdIO(
+                        startupOptions,
+                        CreateKernel(options.DefaultKernel, new BrowserFrontendEnvironment(), startupOptions, null),
```

Comment:
Should we inject the frontend environment?

Reply:
there is no need here to inject, we need to provide one to the create kernel function. That is the composition part for stdio server